include/revng/Model/Filters.h

```diff
@@ -0,0 +1,56 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include "revng/ADT/RecursiveCoroutine.h"
+#include "revng/ADT/UpcastablePointer.h"
+#include "revng/Model/CommonTypeMethods.h"
+#include "revng/Model/Type.h"
+#include "revng/Model/TypeDefinition.h"
+
+//
+// This file provides sample filters to use with containers of model types,
+// for example, to iterate all the scalar arguments of a function, you could use
+//
+// ```cpp
+// for (auto &Scalar : Function.Arguments() | model::filter::Scalar)
+//   do_stuff(Scalar);
+// ```
+//
+// TODO: add new filters here as you find use cases for them.
```

Comment:
There's literally tens of places where we we'd like to do iterate over all `Instruction`s on a `BasicBlock` but just keep `llvm::CallInst` or calls to a specific function. I'd like to give a shot to it one day.

Reply:
Yeah, but this file is for model-specific filters. Those ones would need to get their own home somewhere else :)